cdap-ui/app/cdap/components/AbstractWidget/DLPCustomWidget/DLPRow.tsx

```diff
@@ -0,0 +1,404 @@
+
+    const transforms = this.props.transforms;
+
+    // WidgetProps for inputFieldDropdown
+    const inputFieldProps = {
+      multiselect: true,
+      allowedTypes: [],
+    };
+
+    if (this.state.transform !== '') {
+      if (nestedErrors.length > 0) {
+        this.handleChange({ expanded: true });
+      }
+
+      inputFieldProps.allowedTypes = this.state.transformObj.supportedTypes || [];
+    }
+
+    const shouldShowConfigGroup =
+      this.state.transform !== '' &&
+      Object.keys(this.state.transformProperties).length > 0 &&
+      this.state.expanded &&
+      this.state.transformObj !== null;
+
+    return (
+      <React.Fragment>
+        <div>
+          <div className={this.props.classes.inputContainer}>
+            <span className={this.props.classes.separator}>Apply</span>
+            <FormControl>
+              <Select
+                classes={{ disabled: this.props.classes.disabled }}
+                value={this.state.transform}
+                onChange={this.handleChangeTransformSelect.bind(this, 'transform')}
+                displayEmpty={true}
+                disabled={false}
+              >
+                {transforms.map((option) => {
+                  return (
+                    <MenuItem value={option.name} key={option.name}>
+                      {option.label}
+                    </MenuItem>
+                  );
+                })}
+              </Select>
+            </FormControl>
+            <span className={this.props.classes.separator}>on</span>
+            <MultiSelect
+              disabled={false}
+              value={this.state.filters}
+              widgetProps={{ options: filters }}
+              onChange={this.handleChangeMultiSelect.bind(this, 'filters')}
+            />
+            <span className={this.props.classes.separator}>within</span>
+            <InputFieldDropdown
+              widgetProps={inputFieldProps}
+              value={this.state.fields}
+              onChange={this.handleChangeMultiSelect.bind(this, 'fields')}
+              disabled={false}
+              extraConfig={this.props.extraConfig}
+            />
+          </div>
+
+          {localErrors.map((err) => {
+            try {
+              return (
+                <div key={err.element} className={this.props.classes.errorText}>
+                  {err.msg}
+                </div>
+              );
+            } catch (error) {
+              return null;
+            }
+          })}
+          <div className={this.props.classes.transformContainer}>
+            <If condition={shouldShowConfigGroup}>
+              <ConfigurationGroup
+                pluginProperties={this.configurationGroupPluginProperties}
+                errors={parseNestedErrors(nestedErrors)}
+                values={this.state.transformProperties}
+                classes={this.props.classes}
+                onChange={this.handleChangeTransformOptions.bind(this)}
```

Comment:
Looks like you forgot to remove .bind after your last change?

Reply:
done